File: C++/KartikeyV/Trees/sumOfRootToLeafBinaryNumbers.cpp
void calc(TreeNode* root, int val, int &res){
    if(root==NULL) return;
    if(root->left==NULL and root->right==NULL){
        res+=(val*2+root->val);
        return;
    }
    val = val*2+root->val;
    calc(root->left, val, res);
    calc(root->right, val, res);
}

int solve(TreeNode* root){
    int res = 0;
    calc(root, 0, res);
    return res;
}

File: python/vedant_bhatnagar/DevsnestProblems_THA/Phase_2/Trees/SumOfRootToLeafBinaryNumbers.py
def f(root, sum):
    if root:
        ans = (2*sum) + root.val
        if not root.left and not root.right:
            return ans
        return f(root.left, ans) + f(root.right, ans)
    return 0

def solve(root):
    # CODE HERE
    return f(root, 0)